main.cpp

```diff
@@ -28,510 +28,592 @@ Thing: Car Wash
+
+    if(dragonFirePower >=10 && familyName == "Smaug")
+    {
+        breathFireProbabilityInPercent = 100;
+    }
+    else
+    {
+        breathFireProbabilityInPercent = 0;
+    }
+}
+
+bool Dragon::fightKnight(int courageOfKnight)
+{
+    bool fight;
+    if (courageOfKnight >=10)
+    {
+        return fight = true;
+    }
+    else
+    {
+        return fight = false;
+    }
+}
+//===================================================================
 
 struct HighRise
 {
     struct Lift
-    {          
-        // 5 properties
-//      1) manufacturer's name
+    {
         std::string manufacturerName = "Cibes";
-//      2) lift speed
         float liftSpeedMph = 8.5f;
-//      3) floor number
         int floorNumber = 4;
-//      4) room number
         int roomNumber = 570;
-//      5) number of passengers
         int numberPassengers = 5;
 
-        // 3 actions
-//    1) moves up
-        void moveLiftUp(Lift lift);
-//    2) moves down
-        void moveLiftDown(Lift lift);
-//    3) goes to penthouse suite
-        void goToPenthouse(std::string password);      
+        int moveLift(int floorNumber); // moveLiftUp() and moveLiftDown() consolidated into moveLift()
+        std::string goToPenthouse(std::string password);
     };
 
-//    1) number of lifts
-    int lifts = 3;
-//    2) number of floors
-    int floors = 58;
-//    3) number of rooms
-    int numRooms = 760;
-//    4) hours per week cleaned
+    const int lifts = 3;
+    static const int penthouseFloor = 58;
+    const int numRooms = 760;
     double hoursSpentCleaningPerWeek = 15.4;
-//    5) cost of property
     float costOfProperty = 450005.09f;
 
-//    1) house residents
-    void giveResidentKey(int roomNumber);
-//    2) charge residents
+    void giveResidentKey(int roomNumber, int roomNumberOnContract);
     int invoiceResident(int invoiceAmount);
-//    3) transport residents between floors
-    void transportResident(std::string residentName, Lift lift);
+    int transportResident(int desiredFloor, int currentFloor);
 };
 
-/*
-Thing 3) Brain
-5 properties:
-    1) number of neurones
-    2) number of functioning hemispheres
-    3) memory storage in gigabytes
-    4) calculations per second
-    5) unknown regions
-3 things it can do:
-    1) recognise faces
-    2) estimate distances
-    3) construct a language
- */
+void HighRise::giveResidentKey(int roomNumber, int roomNumberOnContract)
+{
+    if (roomNumber == roomNumberOnContract)
+    {
+        std::cout << "Key dispensed to resident" << std::endl;
+    }
+    else
+    {
+        std::cout << "Key not dispensed! Check invoice paid" << std::endl;
+    }
+}
+
+int HighRise::invoiceResident(int invoiceAmount)
+{
+    return invoiceAmount;
+}
+
+int HighRise::transportResident(int desiredFloor, int currentFloor)
+{
+    Lift lift;
+    if (currentFloor != desiredFloor)
+    {
+        return lift.moveLift(desiredFloor);
+    }
+    else
+    {
+        return desiredFloor;
+    }
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise

Reply:
That's great to know! Thanks Chuck.

Reply:
Done.